src/cargo/ops/cargo_new.rs

```diff
@@ -140,14 +216,66 @@ fn mk(config: &Config, path: &Path, name: &str,
         (None, None, name, None) => name,
     };
 
-    try!(paths::write(&path.join("Cargo.toml"), format!(
+    let license: Option<Vec<License>> = match (&opts.license, cfg.license) {
+        (&None, None) => None,
+        (&Some(ref lic), _) => Some(lic.clone()),
+        (&_, Some(lic)) => Some(lic.clone()),
+    };
+
+    let license_file: Option<String> = match (&opts.license_file, cfg.license_file) {
```

Comment:
These type annotations can likely all be elided (here, above, and below), and it'd be good to reduce the number of `.clone()` or `.to_string()` calls around here as well.